src/util/CancellationHandle.h

```diff
@@ -81,12 +82,13 @@ struct PseudoStopToken {
 class CancellationException : public std::runtime_error {
  public:
   using std::runtime_error::runtime_error;
-  CancellationException(CancellationState reason, std::string_view details)
-      : std::runtime_error{absl::StrCat("Cancelled due to ",
-                                        reason == CancellationState::TIMEOUT
-                                            ? "timeout"
-                                            : "manual cancellation",
-                                        ". Stage: ", details)} {
+  CancellationException(CancellationState reason,
```

Comment:
As I said, maybe we have both a string and a source location (the string being optional).